src/library.js

```diff
@@ -1238,13 +1238,27 @@ mergeInto(LibraryManager.library, {
     return _strptime(buf, format, tm); // no locale support yet
   },
 
+  // Base Emscripten EH error class
+  $EmscriptenEH__docs: '/** @constructor */',
+  $EmscriptenEH: 'function EmscriptenEH() {\n' +
+                 '    this.name = \'EmscriptenEH\';\n' +
```

Comment:
Do we need the name preserved outside of debug mode? (where it won't be minified anyway)

If not, let's not do this to help with code size impact a bit.

Reply:
20fe0a64e9baebf9e519c74d4624742b3cc9792c, we can probably do the same thing for `ExitStatus` as a follow-up.
https://github.com/emscripten-core/emscripten/blob/73ab1c59529692d838125f4b6df87c62d8848fe5/src/library_bootstrap.js#L20
https://github.com/emscripten-core/emscripten/blob/73ab1c59529692d838125f4b6df87c62d8848fe5/src/library_wasi.js#L11

Reply:
Thinking about this further, we could set `this.name` to the thrown C++ type for the `CppException` class. For example, with commit 5514f2fd40ec1387b4674e996c5a8bc7a7526f8c, I now see:

<details>
  <summary>Details</summary>

```console
$ echo -e '#include <stdexcept>\nint main(){throw std::invalid_argument("Invalid argument!");}' | ./emcc -fexceptions -xc++ -o test.js -
$ node test.js
/home/kleisauke/emscripten/test.js:137
      throw ex;
      ^

std::invalid_argument: Invalid argument!
    at ___cxa_throw (/home/kleisauke/emscripten/test.js:1503:13)
    at wasm://wasm/0005dc86:wasm-function[20]:0xaf1
    at wasm://wasm/0005dc86:wasm-function[22]:0xb8c
    at /home/kleisauke/emscripten/test.js:911:22
    at callMain (/home/kleisauke/emscripten/test.js:2267:15)
    at doRun (/home/kleisauke/emscripten/test.js:2321:23)
    at run (/home/kleisauke/emscripten/test.js:2336:5)
    at runCaller (/home/kleisauke/emscripten/test.js:2252:19)
    at removeRunDependency (/home/kleisauke/emscripten/test.js:820:7)
    at receiveInstance (/home/kleisauke/emscripten/test.js:999:5)

Node.js v18.13.0
```

Instead of:
```console
$ node test.js
/home/kleisauke/emscripten/test.js:137
      throw ex;
      ^

CppException [Error]: std::invalid_argument,Invalid argument!
    at ___cxa_throw (/home/kleisauke/emscripten/test.js:1501:13)
    at wasm://wasm/0005dc86:wasm-function[20]:0xaf1
    at wasm://wasm/0005dc86:wasm-function[22]:0xb8c
    at /home/kleisauke/emscripten/test.js:911:22
    at callMain (/home/kleisauke/emscripten/test.js:2265:15)
    at doRun (/home/kleisauke/emscripten/test.js:2319:23)
    at run (/home/kleisauke/emscripten/test.js:2334:5)
    at runCaller (/home/kleisauke/emscripten/test.js:2250:19)
    at removeRunDependency (/home/kleisauke/emscripten/test.js:820:7)
    at receiveInstance (/home/kleisauke/emscripten/test.js:999:5)

Node.js v18.13.0
```
</details>

Reply:
What will it look like when debug info is missing?

Reply:
> What will it look like when debug info is missing?

```bash
$ echo -e '#include <stdexcept>\nint main(){throw std::invalid_argument("Invalid argument!");}' | \
  ./emcc -O3 -sASSERTIONS=0 -sNODEJS_CATCH_EXIT=0 -sNODEJS_CATCH_REJECTION=0 -fexceptions -xc++ -o test.js -
$ ls -lh test.* | awk '{print $5,$9}'
18K test.js
81K test.wasm
# Note: without --unhandled-rejections=warn it'll somehow(?) print the entire minified test.js
# https://github.com/emscripten-core/emscripten/issues/17228
$ node --unhandled-rejections=warn test.js
(node:178719) UnhandledPromiseRejectionWarning: std::invalid_argument: Invalid argument!
    at ___cxa_throw (/home/kleisauke/emscripten/test.js:1:13201)
    at wasm://wasm/00050392:wasm-function[237]:0xf3bb
    at Module._main (/home/kleisauke/emscripten/test.js:1:14829)
    at callMain (/home/kleisauke/emscripten/test.js:1:17192)
    at doRun (/home/kleisauke/emscripten/test.js:1:17589)
    at run (/home/kleisauke/emscripten/test.js:1:17761)
    at runCaller (/home/kleisauke/emscripten/test.js:1:17047)
    at removeRunDependency (/home/kleisauke/emscripten/test.js:1:6121)
    at receiveInstance (/home/kleisauke/emscripten/test.js:1:7770)
    at receiveInstantiationResult (/home/kleisauke/emscripten/test.js:1:7891)
```

Reply:
For reference, here's the output with current ToT:
```bash
$ echo -e '#include <stdexcept>\nint main(){throw std::invalid_argument("Invalid argument!");}' | \
  emcc -O3 -sASSERTIONS=0 -sNODEJS_CATCH_EXIT=0 -sNODEJS_CATCH_REJECTION=0 -fexceptions -xc++ -o test.js -
$ ls -lh test.* | awk '{print $5,$9}'
15K test.js
11K test.wasm
$ node test.js
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "67712".] {
  code: 'ERR_UNHANDLED_REJECTION'
}

Node.js v18.13.0
```

Reply:
Very cool!